### ISSUE TITLE
xbeeboot seems getting stuck in on uno rev3

### ISSUE THREAD

**Issue by gl40** [davidsainty/optiboot#10]
Hi
I'm trying to install this boot load on a rev3 uno. I use another uno as programmer.
Install seems to be OK :

$ avrdude -c avrisp -p atmega328p -P /dev/cu.usbmodem1411 -b19200 -e -u -U lock:w:0x3f:m -U hfuse:w:0xDC:m -U lfuse:w:0xFF:m -U flash:w:xbeeboot_atmega328.hex -U lock:w:0x2f:m 

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.02s

avrdude: Device signature = 0x1e950f (probably m328p)
avrdude: erasing chip
avrdude: reading input file "0x3f"
avrdude: writing lock (1 bytes):

Writing | ################################################## | 100% 0.01s

avrdude: 1 bytes of lock written
avrdude: verifying lock memory against 0x3f:
avrdude: load data lock data from input file 0x3f:
avrdude: input file 0x3f contains 1 bytes
avrdude: reading on-chip lock data:

Reading | ################################################## | 100% 0.01s

avrdude: verifying ...
avrdude: 1 bytes of lock verified
avrdude: reading input file "0xDC"
avrdude: writing hfuse (1 bytes):

Writing | ################################################## | 100% 0.01s

avrdude: 1 bytes of hfuse written
avrdude: verifying hfuse memory against 0xDC:
avrdude: load data hfuse data from input file 0xDC:
avrdude: input file 0xDC contains 1 bytes
avrdude: reading on-chip hfuse data:

Reading | ################################################## | 100% 0.01s

avrdude: verifying ...
avrdude: 1 bytes of hfuse verified
avrdude: reading input file "0xFF"
avrdude: writing lfuse (1 bytes):

Writing | ################################################## | 100% 0.01s

avrdude: 1 bytes of lfuse written
avrdude: verifying lfuse memory against 0xFF:
avrdude: load data lfuse data from input file 0xFF:
avrdude: input file 0xFF contains 1 bytes
avrdude: reading on-chip lfuse data:

Reading | ################################################## | 100% 0.01s

avrdude: verifying ...
avrdude: 1 bytes of lfuse verified
avrdude: reading input file "xbeeboot_atmega328.hex"
avrdude: input file xbeeboot_atmega328.hex auto detected as Intel Hex
avrdude: writing flash (32768 bytes):

Writing | ################################################## | 100% 0.00s

avrdude: 32768 bytes of flash written
avrdude: verifying flash memory against xbeeboot_atmega328.hex:
avrdude: load data flash data from input file xbeeboot_atmega328.hex:
avrdude: input file xbeeboot_atmega328.hex auto detected as Intel Hex
avrdude: input file xbeeboot_atmega328.hex contains 32768 bytes
avrdude: reading on-chip flash data:

Reading | ################################################## | 100% 0.00s

avrdude: verifying ...
avrdude: 32768 bytes of flash verified
avrdude: reading input file "0x2f"
avrdude: writing lock (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of lock written
avrdude: verifying lock memory against 0x2f:
avrdude: load data lock data from input file 0x2f:
avrdude: input file 0x2f contains 1 bytes
avrdude: reading on-chip lock data:

Reading | ################################################## | 100% 0.01s

avrdude: verifying ...
avrdude: 1 bytes of lock verified

avrdude done.  Thank you.

 but after reboot :
-no led flashing
-unable to reset the chip with reset switch.
-unable to access it via ide, needs to burn another boot loader

I've been trying lot of time without any success.
I'am able to write optibot or standard boatload, and it works. 
Any idea on what I'm doing wrong ?
regards

**Comment by davidsainty** [comment]
I think you're programming it ok.  The LED flashing is turned off in the bootloader, to free up the pin.

The default baud rate is 9600bps, which matches the default XBee.  Once it's flashed, try using avrdude at 9600bps and see if that works.

You can also try using the 8MHz .hex file, because with a 16MHz crystal that makes the default baud rate 19200bps :)

**Comment by gl40** [comment]
Thanks for your fast answering, I've been trying lot of settings, without any success :

some captures >>
With xbeeboot_atmega328_pro_8MHz.hex and xbee_atmega328.hex, similar results :
    9600baud:
> avrdude -v  -cstk500v1 -p  atmega328p -P /dev/cu.usbmodem1421    -b 9600
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00
--10 times--

19200bd
>avrdude -v  -cstk500v1 -p  atmega328p -P /dev/cu.usbmodem1421   -b 19200
         SCK period      : 3.3 us
avrdude: AVR device initialized and ready to accept instructions
Reading | ################################################## | 100% 0.02s
avrdude: Device signature = 0x000000 (retrying)
Reading | ################################################## | 100% 0.02s
avrdude: Device signature = 0x000000 (retrying)
Reading | ################################################## | 100% 0.02s
avrdude: Device signature = 0x000000
avrdude: Yikes!  Invalid device signature.

When I try to reboot it with switch nothing append ? is it normal ?

I'm probably wrong somewhere. any idea what I could try   ?

**Comment by gl40** [comment]
I was able to communicate over usb with arduino as programmer and dump the flash
>avrdude -c arduino -p m328p -P /dev/cu.usbmodem1421  -b 9600 -U flash:r:backup.hex:i 

But when I try to do the same OTA, it just loop on "remote at command D3" : 
>>avrdude  -p m328p  -c xbee -P 0013A20041531E4B@/dev/tty.usbserial-A96X9VF7 -U flash:r:backup_xbee.hex  -vvv -b 9600

avrdude: Version 6.3, compiled on Feb  4 2017 at 09:29:16
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "/usr/local/etc/avrdude.conf"
         User configuration file is "/Users/jean/.avrduderc"
         User configuration file does not exist or is not a regular file, skipping

         Using Port                    : 0013A20041531E4B@/dev/tty.usbserial-A96X9VF7
         Using Programmer              : xbee
         Overriding Baud Rate          : 9600
avrdude: Baud 9600
avrdude: Local AT command: AP
avrdude: sendAPIRequest(): 1486504371.091552 -1, 1, -1, 65
avrdude: xbeedev_poll(): Received frame type 88
avrdude: xbeedev_poll(): Local command AP result code 0
avrdude: Remote AT command: D6
avrdude: sendAPIRequest(): 1486504371.122017 -1, 2, -1, 68
avrdude: xbeedev_poll(): Received frame type 97
avrdude: xbeedev_poll(): Remote command 2 result code 0
avrdude: Remote AT command: D3
avrdude: sendAPIRequest(): 1486504371.186048 -1, 2, -1, 68
avrdude: xbeedev_poll(): Received frame type 97
avrdude: xbeedev_poll(): Remote command 3 result code 0
avrdude: Remote AT command: D3
avrdude: sendAPIRequest(): 1486504371.505244 -1, 2, -1, 68
avrdude: xbeedev_poll(): Received frame type 97
...
it look's like a reboot problem, I tried to reboot with button, no success

**Comment by davidsainty** [comment]
Seems like progress.

The D3 command is trying to toggle the Atmel hard reset to get the
bootloader running. But then avrdude should try talking to the bootloader,
which I can't see happening in that log. Can you upload more of the log?
Or is it just hanging at that point?

**Comment by gl40** [comment]
Yes it's definitely progressing :)

I added 0.1microF capacity, just in case.


there the log, without being interrupt, it continues like it forever.
I've absolutely no more idea what to do, so much thanks for you're help 


$ avrdude  -p m328p  -c xbee -P 0013A20041531E4B@/dev/tty.usbserial-A96X9VF7 -U flash:r:backup_xbee.hex  -vvvv

avrdude: Version 6.3, compiled on Feb  4 2017 at 09:29:16
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "/usr/local/etc/avrdude.conf"
         User configuration file is "/Users/jean/.avrduderc"
         User configuration file does not exist or is not a regular file, skipping

         Using Port                    : 0013A20041531E4B@/dev/tty.usbserial-A96X9VF7
         Using Programmer              : xbee
avrdude: XBee address: 0013a20041531e4b
avrdude: Baud 9600
avrdude: Local AT command: AP
avrdude: sendAPIRequest(): 1486741023.247215 -1, 1, -1, 65
avrdude: Send: ~ [7e] . [00] . [05] . [08] . [01] A [41] P [50] . [02] c [63]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [05]
avrdude: Recv: . [88]
avrdude: Recv: . [01]
avrdude: Recv: A [41]
avrdude: Recv: P [50]
avrdude: Recv: . [00]
avrdude: Recv: . [e5]
avrdude: xbeedev_poll(): Received frame type 88
avrdude: xbeedev_poll(): Local command AP result code 0
avrdude: Remote AT command: D6
avrdude: sendAPIRequest(): 1486741023.276754 -1, 2, -1, 68
avrdude: Send: ~ [7e] . [00] . [10] . [17] . [02] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [02] D [44] 6 [36] . [00] . [bb]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [0f]
avrdude: Recv: . [97]
avrdude: Recv: . [02]
avrdude: Recv: . [00]
avrdude: Recv: } [7d]
avrdude: Recv: 3 [33]
avrdude: Recv: . [a2]
avrdude: Recv: . [00]
avrdude: Recv: A [41]
avrdude: Recv: S [53]
avrdude: Recv: . [1e]
avrdude: Recv: K [4b]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: D [44]
avrdude: Recv: 6 [36]
avrdude: Recv: . [00]
avrdude: Recv: 6 [36]
avrdude: xbeedev_poll(): Received frame type 97
avrdude: xbeedev_poll(): Remote command 2 result code 0
avrdude: Remote AT command: D3
avrdude: sendAPIRequest(): 1486741023.340951 -1, 2, -1, 68
avrdude: Send: ~ [7e] . [00] . [10] . [17] . [03] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [02] D [44] 3 [33] . [04] . [b9]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [0f]
avrdude: Recv: . [97]
avrdude: Recv: . [03]
avrdude: Recv: . [00]
avrdude: Recv: } [7d]
avrdude: Recv: 3 [33]
avrdude: Recv: . [a2]
avrdude: Recv: . [00]
avrdude: Recv: A [41]
avrdude: Recv: S [53]
avrdude: Recv: . [1e]
avrdude: Recv: K [4b]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: D [44]
avrdude: Recv: 3 [33]
avrdude: Recv: . [00]
avrdude: Recv: 8 [38]
avrdude: xbeedev_poll(): Received frame type 97
avrdude: xbeedev_poll(): Remote command 3 result code 0
avrdude: Remote AT command: D3
avrdude: sendAPIRequest(): 1486741023.660042 -1, 2, -1, 68
avrdude: Send: ~ [7e] . [00] . [10] . [17] . [04] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [02] D [44] 3 [33] . [05] . [b7]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [0f]
avrdude: Recv: . [97]
avrdude: Recv: . [04]
avrdude: Recv: . [00]
avrdude: Recv: } [7d]
avrdude: Recv: 3 [33]
avrdude: Recv: . [a2]
avrdude: Recv: . [00]
avrdude: Recv: A [41]
avrdude: Recv: S [53]
avrdude: Recv: . [1e]
avrdude: Recv: K [4b]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: D [44]
avrdude: Recv: 3 [33]
avrdude: Recv: . [00]
avrdude: Recv: 7 [37]
avrdude: xbeedev_poll(): Received frame type 97
avrdude: xbeedev_poll(): Remote command 4 result code 0
avrdude: ser_recv(): programmer is not responding
avrdude: sendAPIRequest(): 1486741024.780613 1, 1, 23, 48
avrdude: Send: ~ [7e] . [00] } [7d] 3 [33] . [10] . [05] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [00] . [00] . [01] . [01] . [17] 0 [30]   [20] . [d2]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [07]
avrdude: Recv: . [8b]
avrdude: Recv: . [05]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: k [6b]
avrdude: xbeedev_poll(): Received frame type 8b
avrdude: xbeedev_poll(): Transmit status 5 result code 0
avrdude: ser_recv(): programmer is not responding
avrdude: sendAPIRequest(): 1486741025.846112 1, 1, 23, 48
avrdude: Send: ~ [7e] . [00] } [7d] 3 [33] . [10] . [06] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [00] . [00] . [01] . [01] . [17] 0 [30]   [20] . [d1]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [07]
avrdude: Recv: . [8b]
avrdude: Recv: . [06]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: j [6a]
avrdude: xbeedev_poll(): Received frame type 8b
avrdude: xbeedev_poll(): Transmit status 6 result code 0
avrdude: ser_recv(): programmer is not responding
avrdude: sendAPIRequest(): 1486741026.902616 1, 1, 23, 48
avrdude: Send: ~ [7e] . [00] } [7d] 3 [33] . [10] . [07] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [00] . [00] . [01] . [01] . [17] 0 [30]   [20] . [d0]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [07]
avrdude: Recv: . [8b]
avrdude: Recv: . [07]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: i [69]
avrdude: xbeedev_poll(): Received frame type 8b
avrdude: xbeedev_poll(): Transmit status 7 result code 0
avrdude: ser_recv(): programmer is not responding
avrdude: sendAPIRequest(): 1486741027.961101 1, 1, 23, 48
avrdude: Send: ~ [7e] . [00] } [7d] 3 [33] . [10] . [08] . [00] } [7d] 3 [33] . [a2] . [00] A [41] S [53] . [1e] K [4b] . [ff] . [fe] . [00] . [00] . [01] . [01] . [17] 0 [30]   [20] . [cf]
avrdude: Recv: ~ [7e]
avrdude: Recv: . [00]
avrdude: Recv: . [07]
avrdude: Recv: . [8b]
avrdude: Recv: . [08]
avrdude: Recv: . [0c]
avrdude: Recv: . [f8]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: . [00]
avrdude: Recv: h [68]
avrdude: xbeedev_poll(): Received frame type 8b
avrdude: xbeedev_poll(): Transmit status 8 result code 0
^C

**Comment by gl40** [comment]
Hi there 
It just works ! I don't really know how : I reflashed it (just one more time...). Only change, I did it with another programmer usbasp. Anyway does it change anything ?
Thanks

**Comment by davidsainty** [comment]
Good news!  I can't think why a different programmer would change the
result unless one of the programmers was misbehaving somehow...

On 21/02/2017 10:42 AM, "gl40" <notifications@github.com> wrote:

Hi there
It just works ! I don't really know how : I reflashed it (just one more
time...). Only change, I did it with another programmer usbasp. Anyway does
it change anything ?
Thanks

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/davidsainty/xbeeboot/issues/10#issuecomment-281188252>,
or mute the thread
